### ISSUE TITLE
Was there a change in reg_alpha definition?

### ISSUE THREAD

**Issue by Denisevi4** [dmlc/xgboost#3805]
I used to be using 0.71 version of xgboost. Now when I switched to 0.80 I can't reproduce the model. It appears it's due to reg_alpha.

Code to reproduce the problem with a simple dataset:

    from sklearn.datasets import load_boston
    X, y = load_boston(return_X_y=True)

    import xgboost
    print(xgboost.__version__)
    dtrain = xgboost.DMatrix(X, label=y)
    evals = {}
    params = {'max_depth':3, 'eta':1, 'silent':1, 'objective':'reg:linear', 
          'eval_metric':'rmse', "base_score": 0.0,
          "reg_alpha": 0.1
         }
    bst = xgboost.train(params, dtrain, 100, [(dtrain, 'train')], verbose_eval=False, evals_result=evals)
    print(evals["train"]["rmse"][-1])

I look at the final rmse:

* "reg_alpha": 0.0     -  both versions show 0.175185
* "reg_alpha": 0.1     -  version 0.71 shows 0.189119, 0.80 shows 0.185162
* changing reg_alpha to reg_lambda or gamma and keep it at 0.1 -  both versions show the same result
* adding/changing min_child_weight both versions show the same result

I use CentOs7
Tried both python 2.7 and 3.5

**Comment by RAMitchell** [comment]
@henrygouk could this be a consequence of your changes?

**Comment by henrygouk** [comment]
There have been a couple of fixes since I refactored that bit. Is the problem still present in master?

**Comment by hcho3** [comment]
@henrygouk #3668 added 2x factor to `reg_alpha`. Is it supposed to be there?

**Comment by Denisevi4** [comment]
OK master works fine! I can reproduce the same numbers with a small dataset and with my other small dataset. I'll make a more rigorous test, but it seems fine now. Thanks!

**Comment by hcho3** [comment]
@Denisevi4 That's great to hear. Thanks for reporting the issue.
